README.md

```diff
@@ -752,49 +732,91 @@ Specifies any additional attributes to add to the Valve. Should be a hash of the
 
-*Required if `install_from_source` is set to 'false'.* Specifies the package to install. Valid options: a string containing a valid package name.
+##### `manage_user`
+Specifies whether the user should be managed by this module or not. Default: `$::tomcat::manage_user`
 
-#####`package_options`
+##### `manage_group`
+Specifies whether the group should be managed by this module or not. Default: `$::tomcat::manage_group`
 
-*Unused if `install_from_source` is set to 'true'.* Specify additional options to use on the generated package resource. See the documentation of the [`package` resource type](https://docs.puppetlabs.com/references/latest/type.html#package-attribute-install_options) for possible values.
+##### `package_ensure`
+Determines whether the specified package should be installed. Only valid if `install_from_source` is set to `false`. Maps to the `ensure` parameter of Puppet's native [`package` resource type](https://docs.puppetlabs.com/references/latest/type.html#package). Default: 'present'.
 
-#####`source_strip_first_dir`
+##### `package_name`
+*Required if `install_from_source` is set to `false`.* Specifies the package to install. Valid options: a string containing a valid package name.
 
-Specifies whether to strip the topmost directory of the tarball when unpacking it. Only valid if `install_from_source` is set to 'true'. Valid options: 'true' and 'false'. Default: 'true'.
+##### `package_options`
+*Unused if `install_from_source` is set to `true`.* Specify additional options to use on the generated package resource. See the documentation of the [`package` resource type](https://docs.puppetlabs.com/references/latest/type.html#package-attribute-install_options) for possible values.
 
-#####`source_url`
+#### tomcat::instance
+Declares a tomcat instance.
 
-*Required if `install_from_source` is set to 'true'.* Specifies the source URL to install from. Valid options: a string containing a `puppet://`, `http(s)://`, or `ftp://` URL.
+There are two different modes of use: a single tomcat installation and instance (called "single-instance" by this readme), or a single tomcat installation   with multiple instances, each with its own directory structure (called "multi-instance" by this readme).
 
-##### `user`
+- single-instance: If a `tomcat::instance` is declared with `catalina_home` and `catalina_base` both pointing to the directory of a `tomcat::install` then it only configures a single instance.
+- multi-instance: If a `tomcat::instance` is declared with `catalina_home` pointing to the same directory as a `tomcat::install` and `catalina_base` pointing at a different directory then it is configured as an instance of the Apache Tomcat software. Multiple instances of a single install may be created using this method.
+
+##### `catalina_base`
+Specifies the `$CATALINA_BASE` of the Tomcat instance where logs, configuration files, and the 'webapps' directory are managed. For single-instance installs, this is the same as `$catalina_home`
```

Comment:
`$catalina_home` should be `catalina_home` right? To reference the module param vs the tomcat env variable?